go-controller/pkg/node/node_ip_handler_linux_test.go

```diff
@@ -198,3 +151,88 @@ var _ = Describe("Node IP Handler tests", func() {
 		})
 	})
 })
+
+func configureKubeOVNContext(nodeName string, useNetlink, createNs, createLink bool) *testCtx {
+	node := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+			Annotations: map[string]string{
+				util.OVNNodeHostCIDRs:           `["10.1.1.10/24", "2001:db8::10/64"]`,
+				"k8s.ovn.org/l3-gateway-config": `{"default":{"mac-address":"52:54:00:e2:ed:d0","ip-addresses":["10.1.1.10/24"],"ip-address":"10.1.1.10/24","next-hops":["10.1.1.1"],"next-hop":"10.1.1.1"}}`,
+			},
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{{Address: "10.1.1.10", Type: corev1.NodeInternalIP}},
+		},
+	}
+
+	tc := &testCtx{
+		doneWg:      &sync.WaitGroup{},
+		stopCh:      make(chan struct{}),
+		fakeClient:  fake.NewSimpleClientset(node),
+		mgmtPortIP4: ovntest.MustParseIPNet("10.1.1.2/24"),
+		mgmtPortIP6: ovntest.MustParseIPNet("2001:db8::1/64"),
+	}
+
+	var err error
+	fakeClientset := &util.OVNNodeClientset{
+		KubeClient: tc.fakeClient,
+	}
+	tc.watchFactory, err = factory.NewNodeWatchFactory(fakeClientset, nodeName)
+	Expect(err).NotTo(HaveOccurred())
+	err = tc.watchFactory.Start()
+	Expect(err).NotTo(HaveOccurred())
+
+	fakeMgmtPortConfig := &managementPortConfig{
+		ifName:    nodeName,
+		link:      nil,
+		routerMAC: nil,
+		ipv4: &managementPortIPFamilyConfig{
+			ipt:        nil,
+			allSubnets: nil,
+			ifAddr:     tc.mgmtPortIP4,
+			gwIP:       tc.mgmtPortIP4.IP,
+		},
+		ipv6: &managementPortIPFamilyConfig{
+			ipt:        nil,
+			allSubnets: nil,
+			ifAddr:     tc.mgmtPortIP6,
+			gwIP:       tc.mgmtPortIP6.IP,
+		},
+	}
+
+	fakeBridgeConfiguration := &bridgeConfiguration{bridgeName: "breth0"}
+
+	k := &kube.Kube{KClient: tc.fakeClient}
+	if createNs {
+		tc.ns, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+	}
+	if createLink {
+		setupPrimaryInfFn := func() error {
+			link := ovntest.AddLink("breth0")
+			if err = netlink.AddrAdd(link, &netlink.Addr{IPNet: ovntest.MustParseIPNet("10.1.1.10/24")}); err != nil {
+				return err
+			}
+			return nil
+		}
+		if createNs {
+			Expect(tc.ns.Do(func(netNS ns.NetNS) error {
+				return setupPrimaryInfFn()
+			}))
+		} else {
+			Expect(setupPrimaryInfFn()).ShouldNot(HaveOccurred())
+		}
+	}
+	//FIXME (mk): newAddressManagerInternal calls a sync - when this is removed, remove the need to call that func within NS
```

Comment:
I don't understand this comment very well :( I'll come back to this tomorrow.

Reply:
I explained on slack but for anyone else - a call to `newAddressManagerInternal` performs the following commands:
```
// reproducibility of unit tests.
func newAddressManagerInternal(nodeName string, k kube.Interface, config *managementPortConfig, watchFactory factory.NodeWatchFactory, gwBridge *bridgeConfiguration, useNetlink bool) *addressManager {
	mgr := &addressManager{
		nodeName:       nodeName,
		watchFactory:   watchFactory,
		cidrs:          sets.New[string](),
		mgmtPortConfig: config,
		gatewayBridge:  gwBridge,
		OnChanged:      func() {},
		useNetlink:     useNetlink,
		syncPeriod:     30 * time.Second,
	}
	mgr.nodeAnnotator = kube.NewNodeAnnotator(k, nodeName)
	mgr.sync()  <------Notice
	return mgr
}
```

Notice the `sync()` call. Therefore, to ensure `sync()` exec in the correct netns, we must call `newAddressManagerInternal` in the correct netns.